src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/DataFlowOperationVisitor.cs

```diff
@@ -3757,7 +3757,7 @@ public override TAbstractAnalysisValue VisitIsPattern(IIsPatternOperation operat
 
         public override TAbstractAnalysisValue VisitAwait(IAwaitOperation operation, object? argument)
         {
-            var value = base.VisitAwait(operation, argument);
+            var value = Visit(operation.Operation, argument);
```

Comment:
Need to think a bit more about this one, but if we go with this change, I'd prefer we use the pattern similar to the one here: https://github.com/dotnet/roslyn-analyzers/blob/0a77c2a843c4d77bb97fbddceb8055687858de48/src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/DataFlowOperationVisitor.cs#L3603-L3610

This way, if any custom DFA wants to override the behavior for VisitAwait, they can do so and still get the `TryGetTaskWrappedValue` logic that is executed below.

Reply:
@mavasani I don't quite understand the suggestion. I'm also not sure about `TryGetTaskWrappedValue` or when it returns true. For the tests I added it looks like it returns `false` (which caused `value` to be returned)